src/coreComponents/mesh/generators/CellBlockManager.cpp

```diff
@@ -776,4 +802,398 @@ std::map< string, SortedArray< localIndex > > & CellBlockManager::getNodeSets()
+      break;
+    case 3:
+      static constexpr real64 sqrt5 = 2.2360679774997897;
+      GaussLobattoPts[0] = -1.0;
+      GaussLobattoPts[1] = -1./sqrt5;
+      GaussLobattoPts[2] = 1./sqrt5;
+      GaussLobattoPts[3] = 1.;
+      break;
+    case 4:
+      static constexpr real64 sqrt3_7 = 0.6546536707079771;
+      GaussLobattoPts[0] = -1.0;
+      GaussLobattoPts[1] = -sqrt3_7;
+      GaussLobattoPts[2] = 0.0;
+      GaussLobattoPts[3] = sqrt3_7;
+      GaussLobattoPts[4] = 1.0;
+      break;
+    case 5:
+      static constexpr real64 sqrt__7_plus_2sqrt7__ = 3.50592393273573196;
+      static constexpr real64 sqrt__7_mins_2sqrt7__ = 1.30709501485960033;
+      static constexpr real64 sqrt_inv21 = 0.218217890235992381;
+      GaussLobattoPts[0] = -1.0;
+      GaussLobattoPts[1] = -sqrt_inv21*sqrt__7_plus_2sqrt7__;
+      GaussLobattoPts[2] = -sqrt_inv21*sqrt__7_mins_2sqrt7__;
+      GaussLobattoPts[3] = sqrt_inv21*sqrt__7_mins_2sqrt7__;
+      GaussLobattoPts[4] = sqrt_inv21*sqrt__7_plus_2sqrt7__;
+      GaussLobattoPts[5] = 1.0;
+      break;
+  }
+  return GaussLobattoPts;
+}
+
+static void trilinearInterp( real64 const alpha,
+                             real64 const beta,
+                             real64 const gamma,
+                             real64 const (&X)[8][3],
+                             real64 (& coords)[3] )
+{
+  for( int i=0; i<3; i++ )
+  {
+    coords[i] = X[0][i]*( 1.0-alpha )*( 1.0-beta )*( 1.0-gamma )+
+                X[1][i]*    alpha    *( 1.0-beta )*( 1.0-gamma )+
+                X[2][i]*( 1.0-alpha )*    beta    *( 1.0-gamma )+
+                X[3][i]*    alpha    *    beta    *( 1.0-gamma )+
+                X[4][i]*( 1.0-alpha )*( 1.0-beta )*  gamma+
+                X[5][i]*    alpha    *( 1.0-beta )*  gamma+
+                X[6][i]*( 1.0-alpha )*    beta    *  gamma+
+                X[7][i]*    alpha    *    beta    *  gamma;
+  }
+}
+
+void CellBlockManager::createHighOrderMaps( localIndex const order, MeshLevel const & source, MeshLevel & highOrderMeshLevel )
+{
+
+  // constants for hex mesh
+
+  localIndex const numVerticesPerCell = 8;
+  localIndex const numNodesPerEdge = ( order+1 );
+  localIndex const numNodesPerFace = ( order+1 )*( order+1 );
+  localIndex const numNodesPerCell = ( order+1 )*( order+1 )*( order+1 );
+
+  localIndex const numInternalNodesPerEdge = ( order-1 );
+  localIndex const numInternalNodesPerFace = ( order-1 )*( order-1 );
+  localIndex const numInternalNodesPerCell = ( order-1 )*( order-1 )*( order-1 );
+  localIndex const numLocalVertices = source.getNodeManager().size();
+  localIndex const numLocalEdges = source.getEdgeManager().size();
+  localIndex const numLocalFaces = source.getFaceManager().size();
+
+  localIndex const maxVertexGlobalID = source.getNodeManager().maxGlobalIndex() + 1;
+  localIndex const maxEdgeGlobalID = source.getEdgeManager().maxGlobalIndex() + 1;
+  localIndex const maxFaceGlobalID = source.getFaceManager().maxGlobalIndex() + 1;
```

Comment:
Thanks Thomas for the suggestion. The maxVertexGlobalID, maxEdgeGlobalID, and maxFaceGlobalID are actually obtained at the communications stage. It would be possible to recalculate at CellBlockManager but would be a bit redundant in my opinion. I understand that we don't want to involve MeshLevel (source) at CellBlockManager, so it is passed as input parameters which is probably the simplest way to have them.